### DISCUSSION TITLE
Firmware upgrade from 6.2.3 to current version does not work

### DISCUSSION THREAD

**Discussion by Onyxmoon** [RavenSystem/esp-homekit-devices#2067] [General]
**Detailed steps to reproduce it**
Hey,

I'm not sure if this is an issue or a discussion thing. I have a lamp with firmware 6.2.3. Unfortunately, the update does not work either via the GitHub repository or via my own update server. I have tried the current version as well as 12.4.4, 9.4.1, 7.0.0 and 6.9.0.

The lamp always requests the following files from the server and then nothing happens:
- otaversion
- haaboot.bin.sec
- haaversion
- haamain.bin.sec

Are there any restrictions on which firmware versions can still be updated via OTA? Is the firmware too old? Does it have anything to do with the firmware signature?

Is it possible to solve this without a cable connection? Unfortunately, the circuit board is inaccessible. It was installed via OTA at the time.

Thank you for your help in advance.

**MEPLHAA Script**
```
{
  "a":[{
    "t":30,
    "g":[130,5],
    "s":5
  }]
}
```

**Used device**
LVL Tuya Light with CTT
https://de.aliexpress.com/item/4000115477858.html

**HAA version**
6.2.3

**iOS/iPadOS version**
16.5

**Logs**
Example of update server log:
```
GET /12.4.4/otaversion HTTP/1.1
Host: 10.0.1.225
Range: bytes=0-1

HTTP/1.0 206 Partial Content
Server: aidex/1.3 
Date: Sun, 18 Jun 2023 09:06:15 GMT
Last-Modified: Sun, 18 Jun 2023 08:56:53 GMT
Content-Length: 2
Content-Range: bytes 0-1/2
Connection: close
```

**Screen-shots or videos showing issue**
-

**Comment by shanghaix** [comment]
Try with a different file server,like XAMPP (very friendly to use, no command line required).
I was able to upgrade from 6.0 last week using the custom update server and pointing to the IP address of the machine where XAMPP is running.

**Reply by Onyxmoon** [reply]
Thank you very much. Sometimes the solution is simple. Thank you for the things to consider. It worked well.
